connectivity/netsocket/source/nsapi_dns.cpp

```diff
@@ -982,7 +982,11 @@ static void nsapi_dns_query_async_create(void *ptr)
         }
 
         if (!query->socket_cb_data) {
-            query->socket_cb_data = new SOCKET_CB_DATA;
+            query->socket_cb_data = new (std::nothrow) SOCKET_CB_DATA;
+            if (!query->socket_cb_data) {
+                nsapi_dns_query_async_resp(query, NSAPI_ERROR_NO_MEMORY, NULL);
+                return;
```

Comment:
While I won't pretend I understand this portion too much, the function signature here is `static void nsapi_dns_query_async_create(void *ptr)`. This function is a helper for the async DNS query, and I believe the `nsapi_dns_query_async_resp` call takes care of the error return call.

I did not believe I looked hard enough, however, because I believe that I did not clean up the UDP socket and the appropriate call to the stack created earlier in the same function call.